src/cmd-osbuild

```diff
@@ -267,66 +273,84 @@ main() {
+            else
+                tobuild+=("${platform}")
+                echo "will build $platform"
+                [ "${platform}" == 'qemu' ] && building_qemu=1
+            fi
+        fi
+    done
+    platforms=("${tobuild[@]}")
 
-    # In the jenkins pipelines we build the qemu image first and that operation
-    # will do a lot of the same work required for later artifacts (metal, metal4k, etc)
-    # so we want the cached output from that run to persist. The later artifacts get
-    # built in parallel, so we need to be able to access the cache by multiple processes,
+    # Run OSBuild now to build the platforms that were requested. In the jenkins
+    # pipelines we build the qemu image first and that operation will do a lot of
+    # the same work required for later artifacts (metal, metal4k, etc) so we want
+    # the cached output from that run to persist. For now, later artifacts get built
+    # in parallel, so we need to be able to access the cache by multiple processes,
     # so for those we'll set `snapshot=on` so that each will get their own disk image.
     # This is OK because we don't checkpoint (cache) any of those stages.
-    [ "${platform}" == "qemu" ] && snapshot="off" || snapshot="on"
+    [ "${building_qemu}" == '1' ] && snapshot="off" || snapshot="on"
     runvm_osbuild_config_json="$(generate_runvm_osbuild_config)"
     outdir=$(mktemp -p "${tmp_builddir}" -d)
     runvm_with_cache_snapshot "$snapshot" -- /usr/lib/coreos-assembler/runvm-osbuild                    \
                 --config "${runvm_osbuild_config_json}"                                                 \
                 --mpp "/usr/lib/coreos-assembler/osbuild-manifests/coreos.osbuild.${basearch}.mpp.yaml" \
                 --outdir "${outdir}"                                                                    \
-                --platform "${platform}"
-
-    mv "${outdir}/${platform}/${platform}" "${imgpath}"
-
-    case "$platform" in
-        qemu-secex)
-            # Massage the generated artifact through an extra VM for secex. This
-            # will also create an Ignition pubkey and store it in the meta.json
-            # and builddir.
-            postprocess_qemu_secex
-            # Also need to update the meta.json and builddir with the main artifact.
-            postprocess_artifact "${platform}" "${imgpath}" "${imgname}" 'False'
-            ;;
-        *)
-            # Update the meta.json and builddir with the generated artifact.
-            postprocess_artifact "${platform}" "${imgpath}" "${imgname}" 'False'
-            ;;
-    esac
+                --platforms "$(IFS=,; echo "${platforms[*]}")"
+
+    for platform in "${platforms[@]}"; do
+        suffix=
+        case $platform in
+            applehv|metal|metal4k) suffix=raw ;;
+            gcp) suffix=tar.gz ;;
+            hyperv) suffix=vhdx ;;
+            qemu|qemu-secex) suffix=qcow2 ;;
```

Comment:
I'm confused why we need this. Doesn't osbuild already output the files with the right extension?

Reply:
Previously OSBuild output to whatever filename we told it to. It was a config parameter that was passed in, but that really only worked in the 1:1 coupling we had where we ran OSBuild separately for each platform. In this PR I change the final pipelines for each platform to just output to a filename that matches the platform name (i.e. no suffix).

We could hardcode the suffix inside the OSBuild manifest yaml and maybe we should, but we'd still need to know the suffix here in `cmd-osbuild` in order to copy the file out (i.e. we'd copy `$outdir/$platform/$platform.$suffix` out instead of `$outdir/$platform/$platform`).

So either way this script needs to know what the suffix will be. Whether we add the suffix to the intermediate file generated by OSBuild in the outdir isn't important to me.

Reply:
Right, got it. Hmm, it does seem to me like osbuild should be outputting files with the right extension to make it more self-contained and usable outside of cosa. You're right though that it wouldn't change the fact that this code would also need to know about extensions (unless we use globs I guess, but that feels yucky). Anyway, no need to block on this.

Reply:
> Right, got it. Hmm, it does seem to me like osbuild should be outputting files with the right extension to make it more self-contained and usable outside of cosa.

OK. I found a way to have the actual file names OSBuild generates in the outdir to be the correct filenames we want to store in the builddir. Updated.